bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -349,36 +355,78 @@
 			_t.iconSet = splittedIconAttr[0];
 			_t.iconName = splittedIconAttr[1];
 			if (_t.icon.src !== noneImageSrc) {
+				_t.icon.addEventListener("load", convertToInlineSVG);
 				_t.icon.addEventListener("error", replaceImageWithNone);
 			}
 		}
 
+		_t.replaceIconWithInlineSVG = function(svgText) {
+			var
+				parser,
+				docSvg,
+				newIconElement,
+				dataIcon;
+
+			// Parse the SVG text and turn it into DOM nodes
+			parser = new DOMParser();
+			docSvg = parser.parseFromString(svgText, "image/svg+xml");
+			newIconElement = docSvg.querySelector("svg");
+
+			// Keep the attribute data-icon
+			dataIcon = _t.icon.getAttribute("data-icon");
+			if (dataIcon !== null) {
+				newIconElement.setAttribute("data-icon", dataIcon);
+			}
+
+			// Replace the current icon element with the built inline SVG
+			_t.iconContainer.replaceChild(newIconElement, _t.icon);
+			_t.icon = _t.parentNode.parentNode.querySelector(o.formIconSvg);
+		};
+
+		_t.getSVGIconAndReplaceWithInline = function(srcUrl, checkCurrentColor, defaultSVG) {
+			// fetch(srcUrl, { cache: "force-cache" }).then(function(response) {
+			fetch(srcUrl).then(function(response) {
+				if (response.ok && response.headers.get("content-type") === "image/svg+xml") {
+					response.text().then(function(data) {
+						if (!checkCurrentColor || data.indexOf("currentColor") !== -1) {
+							_t.replaceIconWithInlineSVG(data);
```

Comment:
Instead of `currentColor`, aren’t `Canvas` or `CanvasText` also valid, relative (system) colors, which shall be considered when embedding?

Reply:
Do you mean a canvas can be embedded inside a SVG icon? 
Can you please provide an example?

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/system-color contains a list of so called “system colors”.  With `Canvas` I mean one of these system colors, not a surface for drawing.

That is: `style="background:Canvas;color:CanvasText"` and `style="background:CanvasText;color:Canvas"` are valid in SVG and can be used, when the embedding system does not handle `prefer-color-scheme` or `color-scheme`. ( I have tried the to send emails that are supposed to be rendered in dark mode, when the email client does not understand the notion of dark mode.  I had limited success).

Reply:
So you would like an inline SVG in case the SVG contains Canvas or CanvasText in addition to currentColor?

Reply:
I mean in theory SVGs can contain any “system color“.  It is up to you to handle the system colors, as they are intended.

Whether SVGs are embedded or not, I do not know.

Reply:
But support of system colors is not the purpose of this PR. 
It looks like an additional feature request to be coveted by another PR. 
Please fill a feature request if not yet done.